src/components/exportAndImport/importFromJson.ts

```diff
@@ -33,6 +35,43 @@ const insertFromObject: InsertFromObject = async (
     const recordsForTable = records[tableName]
     if (!recordsForTable || recordsForTable.length === 0) continue
 
+    // When importing a template, check that version doesn't already exist in
+    // the system
+    if (
+      tableName === 'template' &&
+      templates?.checkVersionOnImport &&
+      recordsForTable.length === 1
+    ) {
+      const template = recordsForTable[0]
+
+      // If template was exported using an earlier version schema, we need to
+      // migrate it first
+      if (!('versionId' in template)) {
+        template.versionId = getTemplateVersionId()
+        template.versionComment = 'Migrated from previous version format'
+        template.versionHistory = new Array(template.version).fill(0).map((_) => ({
+          comment: null,
+          timestamp: DateTime.fromISO(template.versionTimestamp),
+          versionId: getTemplateVersionId(),
+          parentVersionId: null,
+        }))
+        template.version = null
+      }
+
+      const existingVersions = await databaseConnect.getTemplateVersionIDs(template.code)
+
+      // Make sure unlocked "*" templates have unique versionId
```

Comment:
Since the "versionId" field must be unique (per template code), if we have more than one "in-progress" version, this just adds a numeric suffix to preserve uniqueness. e.g.
```
*,
*_1,
*_2,
...
```